server/config/validate-config-keys.js

```diff
@@ -37,9 +29,10 @@ const parseConfig = filename => JSON.parse( readConfigFile( filename ) );
 /** @type {Array} list of [ filename, config data keys ] configuration pairs */
 const environmentKeys = fs
 	.readdirSync( configRoot, { encoding: 'utf8' } )
-	.filter( filename => environmentConfigPattern.test( path.basename( filename ) ) )
-	.filter( filename => ! ( /secrets/g ).test( filename ) )
-	.filter( filename => ! ( /client/g ).test( filename ) )
+	.filter( filename => /\.json$/.test( path.basename( filename ) ) ) // only the JSON config files
+	.filter( filename => '_shared.json' !== filename ) // base config for all environments
+	.filter( filename => 'client.json' !== filename ) // whitelist of keys allowed in client
+	.filter( filename => ! ( /secrets/g ).test( filename ) ) // secret tokens not part of this system
```

Comment:
yes to catch `secrets.json` and `empty-secrets.json`

I don't know if we shouldn't actually be linting the same here and leave that REST key blank in `_shared`